File: Removing Stars From a String/removing-stars-from-a-string.java
class Solution {
    public String removeStars(String s) {
        char[] arr = s.toCharArray(); 
        int index = 0;

        for (int i = 0; i < s.length(); i++) {
            if (arr[i] == '*') {
                index--;
            } else {
                arr[index++] = arr[i];
            }
        }

        return new String(arr, 0, index);
    }
}

File: Removing Stars From a String/removing-stars-from-a-string.py
class Solution(object):
    def removeStars(self, s):
        arr = list(s) 
        index = 0
        
        for char in s:
            if char == '*':
                index -= 1
            else:
                arr[index] = char
                index += 1
                
        return ''.join(arr[:index])
